src/plugins/intel_gpu/tests/transformations/reduce_modification_for_false_keepdims_test.cpp

```diff
@@ -0,0 +1,173 @@
+    }
+
+    static std::shared_ptr<ngraph::Function> get_reference_function(const ngraph::PartialShape& input_shape,
+                                                                    const std::vector<int64_t>& axes,
+                                                                    const ReduceType& reduce_type) {
+        auto input = std::make_shared<ngraph::opset10::Parameter>(ngraph::element::f32, input_shape);
+        auto axes_const = ngraph::opset10::Constant::create(ngraph::element::i64, ngraph::Shape{axes.size()}, axes);
+
+        auto reduce = std::make_shared<ngraph::opset10::ReduceMax>(input, axes_const, true);
+
+        return std::make_shared<ngraph::Function>(ngraph::NodeVector{reduce}, ngraph::ParameterVector{input});
+    }
+
+    static std::shared_ptr<ngraph::Function> get_initial_function(const ngraph::PartialShape& input_shape,
+                                                                  const std::vector<int64_t>& axes,
+                                                                  const ReduceType& reduce,
+                                                                  const bool keep_dim,
+                                                                  const bool need_reshape,
+                                                                  const std::vector<size_t>& reshape_shape) {
+        auto param = std::make_shared<ngraph::opset10::Parameter>(ngraph::element::f32, input_shape);
+        ngraph::Output<ngraph::Node> input = param->output(0);
+
+        auto axes_const = ngraph::opset10::Constant::create(ngraph::element::i64, ngraph::Shape{axes.size()}, axes);
+        input = std::make_shared<ngraph::opset10::ReduceMax>(input, axes_const, keep_dim);
+
+        if (need_reshape) {
+            input = std::make_shared<ngraph::opset10::Reshape>(
+                input,
+                ngraph::opset10::Constant::create(ngraph::element::i64,
+                                                  ngraph::Shape{reshape_shape.size()},
+                                                  reshape_shape),
+                false);
+        }
+
+        return std::make_shared<ngraph::Function>(ngraph::NodeVector{input.get_node_shared_ptr()},
+                                                  ngraph::ParameterVector{param});
+    }
+};
+
+TEST_P(ReduceModificationTests, CompareFunctions) {
+    auto unh = std::make_shared<ngraph::pass::UniqueNamesHolder>();
+    ngraph::pass::Manager m;
+    m.register_pass<ngraph::pass::InitUniqueNames>(unh);
+    m.register_pass<ngraph::pass::InitNodeInfo>();
+    m.register_pass<ov::intel_gpu::ModifyReduceForFalseKeepDims>();
+    m.register_pass<ngraph::pass::CheckUniqueNames>(unh);
+    m.run_passes(f);
+    ASSERT_NO_THROW(check_rt_info(f));
+
+    auto fc = FunctionsComparator::no_default().enable(FunctionsComparator::ACCURACY);
+    auto res = fc.compare(f, f_ref);
+    ASSERT_TRUE(res.valid) << res.message;
+}
```

Comment:
I think this test will succeed even if `ModifyReduceForFalseKeepDims` is actually doing nothing. I think we need to confirm that the pass is actually working by counting the number of nodes in the graph.

Reply:
As we discussed, I added a logic to check whether a 'Reshape'.

Reply:
is it intended not to check values of two cases?